apps/documents/tasks.py

```diff
@@ -269,3 +277,82 @@ def delete_document_source_task(self, document_source_id: int):
 
     if not document_source.has_versions:
         document_source.delete()
+
+
+@shared_task(
+    bind=True,
+    base=TaskbadgerTask,
+    acks_late=True,
+    ignore_result=False,
+    autoretry_for=(Exception,),
+    retry_kwargs={"max_retries": 3, "countdown": 60},
+)
+def create_collection_zip_task(self, collection_id: int, team_id: int):
+    """
+    Create a ZIP file containing all manually uploaded files from a collection.
+    """
+    progress_recorder = ProgressRecorder(self)
+
+    try:
+        collection = Collection.objects.get(id=collection_id)
+    except Collection.DoesNotExist:
+        logger.error(f"Collection {collection_id} not found")
+        return None
+
+    # Get all manually uploaded files (excluding document source files)
+    collection_files = CollectionFile.objects.filter(
+        collection=collection, document_source__isnull=True
+    ).select_related("file")
+
+    total_files = collection_files.count()
+
+    if total_files == 0:
+        logger.warning(f"No manually uploaded files found in collection {collection_id}")
+        return None
+
+    zip_buffer = BytesIO()
```

Comment:
This will store the data in memory which could be an issue for large downloads. It would be better to use a temporary file.

If for some reason it doesn't work on prod, we can adjust the infra to include a temp filesystem.